gcloud/datastore/connection.py

```diff
@@ -559,20 +412,22 @@ def _lookup(self, lookup_request, dataset_id, stop_on_deferred):
             _copy_deferred_keys(lookup_request, lookup_response)
         return results, missing, deferred
 
-    def _set_read_options(self, request, eventual):
-        """Validate rules for read options, and assign to the request.
 
-        Helper method for ``lookup()`` and ``run_query``.
-        """
-        transaction = self.transaction()
-        if eventual and transaction:
-            raise ValueError('eventual must be False when in a transaction')
-
-        opts = request.read_options
-        if eventual:
-            opts.read_consistency = datastore_pb.ReadOptions.EVENTUAL
-        elif transaction:
-            opts.transaction = transaction.id
+def _set_read_options(request, eventual):
+    """Validate rules for read options, and assign to the request.
+
+    Helper method for ``lookup()`` and ``run_query``.
+    """
+    current = Batch.current()
+    transaction = isinstance(current, Transaction) and current or None
```

Comment:
I see the difficulty / convenience of using the current transaction.

Doing a quick `git grep '\.lookup'` (and similar for `run_query`) they are only used once, so I think it's fine to leave it up to the callers to pass through the transaction information. Acceptiong `transaction_id=None` is probably sufficient for this method?